src/yamerl_node_map.erl

```diff
@@ -141,16 +141,15 @@ node_as_proplist_or_map(#yamerl_constr{ext_options = Options}) ->
 
 new_builder(Constr) ->
     Format = node_as_proplist_or_map(Constr),
-    Map = #map_builder{
```

Comment:
`data` is defined as `proplists:proplist() | map()`, so making temporary `Map` leaving `data` undefined was illegal.